libs/eo/transfers/src/lib/eo-transfers-modal.component.ts

```diff
@@ -0,0 +1,136 @@
+import { WATT_FORM_FIELD } from '@energinet-datahub/watt/form-field';
+import { WattInputDirective } from '@energinet-datahub/watt/input';
+import { WATT_MODAL, WattModalComponent } from '@energinet-datahub/watt/modal';
+import { WattRangeValidators } from '@energinet-datahub/watt/validators';
+import { EoTransfersService } from './eo-transfers.service';
+import { EoTransfersStore } from './eo-transfers.store';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'eo-transfers-modal',
+  imports: [
+    WATT_MODAL,
+    WATT_FORM_FIELD,
+    WattButtonComponent,
+    FormsModule,
+    ReactiveFormsModule,
+    WattInputDirective,
+    WattDatepickerComponent,
+    NgIf,
+    JsonPipe,
+  ],
+  standalone: true,
+  styles: [
+    `
+      watt-form-field {
+        margin-top: var(--watt-space-l);
+        margin-bottom: var(--watt-space-m);
+      }
+    `,
+  ],
+  template: `<watt-modal
+    #modal
+    [title]="title"
+    size="small"
+    closeLabel="Close modal"
+    [loading]="requestLoading"
+  >
+    <form [formGroup]="form">
+      <watt-form-field>
+        <watt-label>Receiver</watt-label>
+        <input wattInput required="true" type="text" formControlName="tin" [maxlength]="8" />
+        <watt-error *ngIf="form.controls.tin.errors"
+          >An 8-digit TIN/CVR number is required</watt-error
+        >
+      </watt-form-field>
+      <watt-form-field>
+        <watt-label>Period</watt-label>
+        <watt-datepicker required="true" formControlName="dateRange" [range]="true" />
+        <watt-error *ngIf="form.controls.dateRange.errors"> A date range is required </watt-error>
+      </watt-form-field>
+    </form>
+    <watt-modal-actions>
+      <watt-button variant="secondary" (click)="modal.close(false)">Close</watt-button>
+      <watt-button [disabled]="!form.valid" (click)="createAgreement()"
+        >Create transfer agreement</watt-button
+      >
+    </watt-modal-actions>
+  </watt-modal>`,
+})
+export class EoTransfersModalComponent {
+  @ViewChild(WattModalComponent) modal!: WattModalComponent;
+  @Input() title = '';
+
+  form = new FormGroup({
+    tin: new FormControl('', [Validators.minLength(8), Validators.pattern('^[0-9]*$')]),
+    dateRange: new FormControl<WattRange>({ start: '', end: '' }, [WattRangeValidators.required()]),
+  });
+  requestLoading = false;
+
+  constructor(
+    private service: EoTransfersService,
+    private store: EoTransfersStore,
+    private cd: ChangeDetectorRef
+  ) {}
+
+  open() {
+    this.modal.open();
+  }
+
+  createAgreement() {
+    if (!this.form.controls['dateRange'].value || !this.form.controls['tin'].value) return;
+
+    const transfer = {
+      startDate: Math.round(new Date(this.form.controls['dateRange'].value.start).getTime() / 1000),
+      endDate: Math.round(new Date(this.form.controls['dateRange'].value.end).getTime() / 1000 - 1), // -1 to offset that endDate is next day at 00:00, when we need chosen day at 23:59:59
```

Comment:
I completely agree. We usually have talked about that "end dates" should be until (but not including) the last hour/time that was sent forward. So if i send a time that says it should end at 12 feb, i will send that it ends at 12 feb 00:00 UTC. Unfortunately the current implementation of the API then sends back to me that the end date is also at this time. That means when i ask for "what day is the end date", it will says 12 feb., but in reality it ends on 11 feb 23:59:59.

So either i subtract the 1 second there, or i then always format dates to show "the date - 1 day". Which do you prefer? :)

Reply:
Do not subtract 1 from these times. We will get into a world of problems. Here you are subtracting a single second - somewhere else it is a millisecond, somewhere else it is a minute; it ends up giving a lot of issues. 
Think of the time for a certificate this way: Each certificate is hourly-based, so a certificate that starts at 23:00:00 lasts 60 minutes; 23:00:00 + 60 minutes means that it will end at 24:00:00 (or 0:00:00). 

If there is a bug in the API, then you have to take that up with Saiuan or Christoffer.

Reply:
Since the API and design currently don't support setting hours, and it seems like the datepicker wants to use start of day on startDate and end of day on endDate, i opted for just compensating when showing the table data :) It just goes back one day from whatever date it has set as endDate.

Reply:
The API allows any timestamp you give it, so it can accept a period of `2023-06-12 01:23:45` to `2023-06-14 12:34:56`.

UI wise this is simplified to using a DatePicker. I have talked with Søren PO and what he wants is the start date and end date selected in the date picker is inclusive. This mean, that if the user picks `2023-06-12` as start and `2023-06-14`, then it should result in an agreement running from `2023-06-12 00:00` in the browser's timezone to `2023-06-15 00:00` in the browser's timezone. Remember that when communicating with the API, then the timestamps will always be in UTC (UnixTime Seconds) and the UI have to do the mapping between the timestamps in browser's timezone and UTC.

Reply:
We found a solution to this :) FE should show the date from the day before, if the end date is midnight of the day after